documentation/source/directives.rst

```diff
@@ -34,15 +34,15 @@ The available directives are shown below. In each case the ``project``,
    should be used for this directive. This overrides the default project if one
    has been specified.
 
-   This is not used by the ``autodoxygenindex`` directive. Use ``source``
-   instead to specify the entry in the ``breathe_projects_source`` config value
+   This is not used by the `autodoxygenindex`_ directive. Use ``source``
```

Comment:
there's no rush here. I'll look into that (never used `:rst:dir:` before)

Reply:
Well, that's not a trivial change... This idea may take me a while because of all the documented directives and corresponding options (some of which are listed as a table - not sure how to address that).

### Question
Should the RST domain be used on the detailed directive examples?
Or should it be used on the directives.rst doc?

My initial reaction (because it would be less work) is to use the RST domain on the corresponding directive example pages and keep using the arbitrary reference (like the one pointed out here) to the general directive descriptions' section header (in directives.rst).

Of course, I might change my mind when I start playing around with the RST domain.

Reply:
Good question. From a brief look I think this should be postponed to another PR that restructures the docs. It seems that there is some duplication between the detailed directive pages, and then the overview page.
My immediate though is that ``directives.rst`` should have a list of directives, with just a single-sentence description and a ``:rst:dir:`` to the actual documentation of it. Then the individual pages should use the rst domain features to fully document it.

Reply:
See #822